### PR TITLE
docs: Examples for column and table classes

### LINKED ISSUE #224: Examples for `Table` and `Column` class
The docstrings of the `Row` class already contain `Examples` blocks. These should be added for the `Table` and `Column` classes as well.

### PR DESCRIPTION
**Pull request by ilkajw**
Closes #224 .

### Summary of Changes

Added examples to the docstrings of all methods of the classes Table and Column.

### CODE REVIEW

**Comment by lars-reimann** [comment]
We need to wait until [GitHub Actions is back up](https://www.githubstatus.com/). In the meantime I've merge `main` into this branch. Please check whether all functions of `Table` and `Column` are now covered by docstrings. You can also locally test whether the examples are correct by running

```
poetry run pytest --doctest-modules
```

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/6.22.2) status: ❌ [ERROR](https://github.com/Safe-DS/Stdlib/actions/runs/4944530494)

| Descriptor  |                                 Linter                                 |Files|Fixed|                            Errors                            |Elapsed time|
|-------------|------------------------------------------------------------------------|-----|----:|--------------------------------------------------------------|------------|
|✅ PYTHON    |[black](https://megalinter.io/6.22.2/descriptors/python_black)          |    2|    1|                                                             0|1.56s       |
|✅ PYTHON    |[mypy](https://megalinter.io/6.22.2/descriptors/python_mypy)            |    2|     |                                                             0|2.24s       |
|❌ PYTHON    |[ruff](https://megalinter.io/6.22.2/descriptors/python_ruff)            |    2|    1|[9](https://github.com/Safe-DS/Stdlib/actions/runs/4944530494)|0.08s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/6.22.2/descriptors/repository_git_diff)|yes  |     |no                                                            |0.03s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/4944530494)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: src/safeds/data/tabular/containers/_table.py [hunk 1]
```diff
@@ -1191,6 +1588,16 @@ def split(self, percentage_in_first: float) -> tuple[Table, Table]:
...
+
+        Examples
+        --------
+        >>> from safeds.data.tabular.containers import Table
+        >>> table = Table.from_dict({"temperature": [10, 15, 20, 25, 30], "sales": [54, 74, 90, 206, 210]})
+        >>> slices = table.split(0.4)
+        >>> slices[0].to_dict()
+        {'temperature': [10, 15], 'sales': [54, 74]}
+        >>> slices[1].to_dict()
+        {'temperature': [20, 25, 30], 'sales': [90, 206, 210]}
```
Comment (alex-senger) [inline src/safeds/data/tabular/containers/_table.py:1600]: Why are you casting the slices into dicts?

File: src/safeds/data/tabular/containers/_table.py [hunk 2]
```diff
@@ -255,6 +288,18 @@ def from_rows(rows: list[Row]) -> Table:
...
             If any of the row schemas does not match with the others.
+
+        Examples
+        --------
+        >>> from safeds.data.tabular.containers import Table
+        >>> from safeds.data.tabular.containers import Row
+        >>> row1 = Row({"a": 1, "b": 2})
+        >>> row2 = Row({"a": 3, "b": 4})
+        >>> table = Table.from_rows([row1, row2])
+        >>> table
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_table.py:299]:
```suggestion
        >>> Table.from_rows([row1, row2])
```

File: src/safeds/data/tabular/containers/_table.py [hunk 3]
```diff
@@ -218,8 +238,21 @@ def from_columns(columns: list[Column]) -> Table:
...
             If multiple columns have the same name.
+
+        Examples
+        --------
+        >>> from safeds.data.tabular.containers import Table
+        >>> from safeds.data.tabular.containers import Column
+        >>> col1 = Column("a", [1, 2, 3])
+        >>> col2 = Column("b", [4, 5, 6])
+        >>> table = Table.from_columns([col1, col2])
+        >>> table
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_table.py:251]:
```suggestion
        >>> Table.from_columns([col1, col2])
```

File: src/safeds/data/tabular/containers/_table.py [hunk 4]
```diff
@@ -196,6 +206,16 @@ def from_dict(data: dict[str, list[Any]]) -> Table:
         ------
         ColumnLengthMismatchError
             If columns have different lengths.
+
+        Examples
+        --------
+        >>> from safeds.data.tabular.containers import Table
+        >>> d = {'a': [1, 2], 'b': [3, 4]}
+        >>> table = Table.from_dict(d)
+        >>> table
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_table.py:215]:
```suggestion
        >>> Table.from_dict(d)
```

File: src/safeds/data/tabular/containers/_table.py [hunk 5]
```diff
@@ -1225,6 +1632,13 @@ def tag_columns(self, target_name: str, feature_names: list[str] | None = None)
             If the target column is also a feature column.
         ValueError
             If no feature columns are specified.
+
+        Examples
+        --------
+        >>> from safeds.data.tabular.containers._table import Table
+        >>> from safeds.data.tabular.containers._tagged_table import Table
+        >>> table = Table.from_dict({"item": ["apple", "milk", "beer"], "price": [1.10, 1.19, 1.79], "amount_bought": [74, 72, 51]})
+        >>> tagged_table = table.tag_columns("amount_bought")
```
Comment (alex-senger) [inline src/safeds/data/tabular/containers/_table.py:1641]:
```suggestion
        >>> tagged_table = table.tag_columns(target_name="amount_bought", feature_names=["apple", "milk", "beer"])
```

**Review by Marsmaennchen221 — CHANGES_REQUESTED** [review]

File: src/safeds/data/tabular/containers/_table.py [hunk 6]
```diff
@@ -1246,6 +1660,13 @@ def transform_column(self, name: str, transformer: Callable[[Row], Any]) -> Tabl
         UnknownColumnNameError
             If the column does not exist.
 
+        Examples
+        --------
+        >>> from safeds.data.tabular.containers import Table
+        >>> table = Table.from_dict({"item": ["apple", "milk", "beer"], "price": [1.00, 1.19, 1.79]})
+        >>> cents = table.transform_column("price", lambda row: row.get_value("price") * 100)
+        >>> cents.to_dict()
+        {'item': ['apple', 'milk', 'beer'], 'price': [100.0, 119.0, 179.0]}
```
Comment (alex-senger) [inline src/safeds/data/tabular/containers/_table.py:1669]: Just returning the Table would be better imo.

File: src/safeds/data/tabular/containers/_table.py [hunk 7]
```diff
@@ -1279,13 +1700,13 @@ def transform_table(self, transformer: TableTransformer) -> Table:
...
-        0      1.0      0.0      1.0      0.0      0.0
-        1      0.0      1.0      0.0      1.0      0.0
-        2      1.0      0.0      0.0      0.0      1.0
+        >>> table = Table.from_dict({"a": [1, 2, 1], "b": [1, 2, 4]})
+        >>> transformer = transformer.fit(table, None)
+        >>> table.transform_table(transformer)
+           a__1  a__2  b__1  b__2  b__4
+        0   1.0   0.0   1.0   0.0   0.0
+        1   0.0   1.0   0.0   1.0   0.0
+        2   1.0   0.0   0.0   0.0   1.0
```
Comment (alex-senger) [inline src/safeds/data/tabular/containers/_table.py:1709]: Not a good example, as the OneHotEncoder shouldn't really be used on numerical values.

File: src/safeds/data/tabular/containers/_column.py [hunk 8]
```diff
@@ -637,7 +957,7 @@ def plot_histogram(self) -> Image:
     # ------------------------------------------------------------------------------------------------------------------
 
     def to_html(self) -> str:
-        """
+        r"""
```
[the remark this answers is not in the document]
Comment (patrikguempel) [reply src/safeds/data/tabular/containers/_column.py:960]: Yes, as backslashes are used within the example's return value

**Review by alex-senger — CHANGES_REQUESTED** [review]

File: src/safeds/data/tabular/containers/_table.py [hunk 9]
```diff
@@ -954,6 +1242,29 @@ def remove_rows_with_outliers(self) -> Table:
...
+        Examples
+        --------
+        >>> from safeds.data.tabular.containers import Column, Table
+        >>> c1 = Column("a", [1, 3, 1, 0.1, 0, 0, 0, 0, 0, 0, 0, 0])
+        >>> c2 = Column("b", [1.5, 1, 0.5, 0.01, 0, 0, 0, 0, 0, 0, 0, 0])
+        >>> c3 = Column("c", [0.1, 0.00, 0.4, 0.2, 0, 0, 0, 0, 0, 0, 0, 0])
+        >>> c4 = Column("d", [-1000000, 1000000, -1000000, -1000000, -1000000, -1000000, -1000000, -1000000, -1000000, -1000000, -1000000, -1000000])
+        >>> table = Table.from_columns([c1, c2, c3, c4])
+        >>> new_table = table.remove_rows_with_outliers()
+        >>> new_table
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_table.py:1255]:
```suggestion
        >>> table.remove_rows_with_outliers()
```

File: src/safeds/data/tabular/containers/_column.py [hunk 10]
```diff
@@ -590,6 +900,13 @@ def plot_boxplot(self) -> Image:
         ------
         NonNumericColumnError
             If the data contains non-numerical data.
+
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_column.py:903]:
```suggestion
```

File: src/safeds/data/tabular/containers/_column.py [hunk 11]
```diff
@@ -661,13 +985,21 @@ def to_html(self) -> str:
...
     def _repr_html_(self) -> str:
-        """
+        r"""
         Return an HTML representation of the column.
 
         Returns
         -------
         output : str
             The generated HTML.
+
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_column.py:995]:
```suggestion
```

File: src/safeds/data/tabular/containers/_table.py [hunk 12]
```diff
@@ -400,6 +504,14 @@ def number_of_columns(self) -> int:
         -------
         number_of_columns : int
             The number of columns.
+
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_table.py:507]:
```suggestion
```

File: src/safeds/data/tabular/containers/_table.py [hunk 13]
```diff
@@ -517,6 +673,17 @@ def get_row(self, index: int) -> Row:
         ------
         IndexOutOfBoundsError
             If no row at the specified index exists in this table.
+
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_table.py:676]:
```suggestion
```

File: src/safeds/data/tabular/containers/_table.py [hunk 14]
```diff
@@ -893,6 +1146,16 @@ def remove_columns_with_missing_values(self) -> Table:
         -------
         table : Table
             A table without the columns that contain missing values.
+
+
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_table.py:1150]:
```suggestion
```

File: src/safeds/data/tabular/containers/_table.py [hunk 15]
```diff
@@ -987,6 +1298,15 @@ def rename_column(self, old_name: str, new_name: str) -> Table:
             If the specified old target column name does not exist.
         DuplicateColumnNameError
             If the specified new target column name already exists.
+
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_table.py:1301]:
```suggestion
```

File: src/safeds/data/tabular/containers/_table.py [hunk 16]
```diff
@@ -1136,6 +1485,21 @@ def sort_columns(
...
+        >>> from safeds.data.tabular.containers import Table
+        >>> table = Table.from_dict({"a": [1], "b": [2] })
+        >>> table.sort_columns(lambda col1, col2: 1)
+           a  b
+        0  1  2
+        >>> table.sort_columns(lambda col1, col2: -1)
+           b  a
+        0  2  1
+        >>> table = Table.from_dict({"b": [2], "a": [1]})
+        >>> table.sort_columns()
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_table.py:1500]:
```suggestion
        >>> table2 = Table.from_dict({"b": [2], "a": [1]})
        >>> table2.sort_columns()
```

**Review by Marsmaennchen221 — CHANGES_REQUESTED** [review]

**Review by Marsmaennchen221 — DISMISSED** [review]

**Review by alex-senger — APPROVED** [review]
[LGTM](https://lgtm.party/p/VAC64N8/)

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**